build-support/bin/generate_docs.py

```diff
@@ -424,7 +424,9 @@ def process_targets_input(cls, help_info: dict[str, Any]) -> dict[str, dict[str,
                     "required" if field["required"] else f"default: <code>{default_str}</code>"
                 )
                 field["description"] = str(field["description"])
-            target["fields"] = sorted(target["fields"], key=lambda fld: cast(str, fld["alias"]))
+            target["fields"] = sorted(
+                target["fields"], key=lambda fld: (-fld["required"], cast(str, fld["alias"]))
```

Comment:
What does the `-` do?

Reply:
My guess it reverses the effective sort order.
Required being a bool, with a unary minus, it is turned into an int (news to me! :P )

![image](https://user-images.githubusercontent.com/72965/169783282-1aa5f655-d844-476e-bc7c-5b6b54081469.png)